eureka2-client/src/main/java/com/netflix/eureka2/client/resolver/EurekaRemoteResolverStep.java

```diff
@@ -0,0 +1,23 @@
+package com.netflix.eureka2.client.resolver;
+
+import com.netflix.eureka2.interests.Interest;
+import com.netflix.eureka2.registry.instance.InstanceInfo;
+
+/**
+ * @author David Liu
+ */
+public interface EurekaRemoteResolverStep {
+
+    /**
+     * optimization for common usecases
+     */
+    public ServerResolver forApps(String... appNames);
```

Comment:
I would drop all forApps, forVips methods here in other places. We should use Interests.forXXX as a first class citizen, so we do not repeat ourselves.